internal/builders/container/README.md

```diff
@@ -0,0 +1,255 @@
+# Generation of SLSA3+ provenance for container images
+
+This document explains how to generate SLSA provenance for container images.
+
+This can be done by adding an additional step to your existing Github Actions
+workflow to call a [reusable
+workflow](https://docs.github.com/en/actions/using-workflows/reusing-workflows)
+to generate generic SLSA provenance. We'll call this workflow the "container
+workflow" from now on.
+
+The container workflow differs from ecosystem specific builders (like the [Go
+builder](../go)) which build the artifacts as well as generate provenance. This
+project simply generates provenance as a separate step in an existing workflow.
+
+---
+
+- [Project Status](#project-status)
+- [Benefits of Provenance](#benefits-of-provenance)
+- [Generating Provenance](#generating-provenance)
+  - [Getting Started](#getting-started)
+  - [Supported Triggers](#supported-triggers)
+  - [Workflow Inputs](#workflow-inputs)
+  - [Provenance Format](#provenance-format)
+  - [Provenance Example](#provenance-example)
+
+---
+
+## Project Status
+
+This workflow is currently under active development. The API could change while
+approaching an initial release.
+
+## Benefits of Provenance
+
+Using the container workflow will generate a non-forgeable attestation to the
+container image using the identity of the GitHub workflow. This can be used
+to create a positive attestation to a container image coming from your
+repository.
+
+That means that once your users verify the image they have downloaded they
+can be sure that the image was created by your repository's workflow and
+hasn't been tampered with.
+
+## Generating Provenance
+
+The container workflow uses a Github Actions reusable workflow to generate the
+provenance.
+
+### Getting Started
+
+To get started, you will need to add some steps to your current workflow. We
+will assume you have an existing Github Actions workflow to build your project.
+
+```yaml
+provenance:
+  needs: [build]
+  permissions:
+    actions: read # for detecting the Github Actions environment.
+    id-token: write # for creating OCID tokens for signing.
+    packages: write # for uploading attestations.
+  if: startsWith(github.ref, 'refs/tags/')
+  # TODO(https://github.com/slsa-framework/slsa-github-generator/issues/492): Use a tagged release once we have one.
+  uses: slsa-framework/slsa-github-generator/.github/workflows/generator_container_slsa3.yml@main
+  with:
+    image: ${{ needs.build.outputs.tag }}
+    registry-username: ${{ github.actor }}
+    # TODO(https://github.com/slsa-framework/slsa-github-generator/issues/492): Remove after GA release.
+    compile-generator: true
```

Comment:
remove `compile-generator` from example?

Reply:
Until this is GA, the example won't pass verification, so it's fine to remove it, I'd say.
But I see what you mean. SO long as we don't forget to update the doc after GA, that's fine with me.